napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1521 @@
+                self._is_selecting = False
+                self._set_highlight()
+            self._is_moving = False
+            self._drag_start = None
+            self._drag_box = None
+            self._fixed_vertex = None
+            self._moving_shape = None
+            self._moving_vertex = None
+            self._hover_shape = shape
+            self._hover_vertex = shape
+            self._set_highlight()
+            self.status = self.get_message(coord, shape, vertex)
+        elif (self.mode == 'add_rectangle' or self.mode == 'add_ellipse' or
+                self.mode == 'add_line'):
+            self._finish_drawing()
+            shape, vertex = self._shape_at(coord)
+            self.status = self.get_message(coord, shape, vertex)
+        elif (self.mode == 'add_path' or self.mode == 'add_polygon'):
+            pass
+        elif (self.mode == 'vertex_insert' or self.mode == 'vertex_remove'):
+            pass
+        else:
+            raise ValueError("Mode not recongnized")
+
+    def on_key_press(self, event):
+        """Called whenever key pressed in canvas.
+
+        Parameters
+        ----------
+        event : Event
+            Vispy event
+        """
+        if event.native.isAutoRepeat():
+            return
+        else:
+            if event.key == ' ':
+                if self.mode != 'pan/zoom':
+                    self._mode_history = self.mode
+                    self._selected_shapes_history = copy(self.selected_shapes)
+                    self.mode = 'pan/zoom'
+                else:
+                    self._mode_history = 'pan/zoom'
+            elif event.key == 'Shift':
+                self._fixed_aspect = True
+                box = self._selected_box
+                if box is not None:
+                    size = box[4]-box[0]
+                    if not np.any(size == np.zeros(2)):
+                        self._aspect_ratio = abs(size[1] / size[0])
+                    else:
+                        self._aspect_ratio = 1
+                else:
+                    self._aspect_ratio = 1
+                if self._is_moving:
+                    self._move(self._cursor_coord)
+            elif event.key == 'r':
+                self.mode = 'add_rectangle'
+            elif event.key == 'e':
+                self.mode = 'add_ellipse'
+            elif event.key == 'l':
+                self.mode = 'add_line'
+            elif event.key == 't':
+                self.mode = 'add_path'
+            elif event.key == 'p':
+                self.mode = 'add_polygon'
+            elif event.key == 'd':
+                self.mode = 'direct'
+            elif event.key == 's':
+                self.mode = 'select'
+            elif event.key == 'z':
+                self.mode = 'pan/zoom'
+            elif event.key == 'v':
+                self.mode = 'vertex_insert'
+            elif event.key == 'x':
+                self.mode = 'vertex_remove'
```

Comment:
These should be in a config file (I favour yaml) and loaded, rather than hardcoded here...

Reply:
Can we do this in a follow up PR? I might need help from @kne42